catalog/spatial/registry/registry-schema-bindings/src/main/java/org/codice/ddf/registry/schemabindings/converter/type/RegistryObjectTypeConverter.java

```diff
@@ -0,0 +1,237 @@
+     * InternationalStringTypeHelper
+     *
+     * @param map the Map representation of the RegistryObjectType to generate, null returns empty Optional
+     * @return Optional RegistryObjectType created from the values in the map
+     */
+    public Optional<RegistryObjectType> convertRegistryObject(Map<String, Object> map) {
+        Optional<RegistryObjectType> optionalRegistryObject = Optional.empty();
+
+        if (map.containsKey(CLASSIFICATION_KEY)) {
+            List<Map<String, Object>> classificationMaps = (List<Map<String, Object>>) map.get(
+                    CLASSIFICATION_KEY);
+
+            ClassificationTypeConverter classificationConverter = new ClassificationTypeConverter();
+
+            Optional<ClassificationType> optionalClassification;
+            for (Map<String, Object> classificationMap : classificationMaps) {
+                optionalClassification = classificationConverter.convert(classificationMap);
+                if (optionalClassification.isPresent()) {
+                    if (!optionalRegistryObject.isPresent()) {
+                        optionalRegistryObject = Optional.of(createObjectInstance());
+                    }
+
+                    optionalRegistryObject.get()
+                            .getClassification()
+                            .add(optionalClassification.get());
+                }
+            }
+        }
+
+        String valueToPopulate = MapUtils.getString(map, DESCRIPTION_KEY);
+        if (StringUtils.isNotBlank(valueToPopulate)) {
+            if (!optionalRegistryObject.isPresent()) {
+                optionalRegistryObject = Optional.of(createObjectInstance());
+            }
+
+            InternationalStringType istToPopulate = INTERNATIONAL_STRING_TYPE_HELPER.create(
+                    valueToPopulate);
+            if (istToPopulate != null) {
+                optionalRegistryObject.get()
+                        .setDescription(istToPopulate);
+            }
+        }
+
+        if (map.containsKey(EXTERNAL_IDENTIFIER_KEY)) {
+            Optional<ExternalIdentifierType> optionalExternalIdentifier;
+            ExternalIdentifierTypeConverter eitConverter = new ExternalIdentifierTypeConverter();
+            for (Map<String, Object> externalIdentifierMap : (List<Map<String, Object>>) map.get(
+                    EXTERNAL_IDENTIFIER_KEY)) {
+                optionalExternalIdentifier = eitConverter.convert(externalIdentifierMap);
+
+                if (optionalExternalIdentifier.isPresent()) {
+                    if (!optionalRegistryObject.isPresent()) {
+                        optionalRegistryObject = Optional.of(createObjectInstance());
+                    }
+
+                    optionalRegistryObject.get()
+                            .getExternalIdentifier()
+                            .add(optionalExternalIdentifier.get());
+                }
+
+            }
+        }
+
+        valueToPopulate = MapUtils.getString(map, HOME_KEY);
```

Comment:
This code is repeated everywhere. We should be able to create a small utility class that can be customized to use the proper RIM_FACTORY method

```
public class MapToSchemaElement<T> {
    private final T objectFactory;

    public MapToSchemaElement(Supplier<T> objectFactory) {
        this.objectFactory = objectFactory;
    }

    public void populateStringElement(Map<String, Object> map, String mapKey, Optional<T> element, Consumer<String> updater) {
        String valueToPopulate = MapUtils.getString(map, mapKey);
        if (StringUtils.isNotBlank(valueToPopulate)) {
            if (!element.isPresent()) {
                element = Optional.of(objectFactory.get());
            }

            updater.accept(valueToPopulate);
        }
    }
}

public class RegistryObjectTypeConverter {
    private final MapToSchemaElement<RegistryObjectType> mapToSchemaElement = new MapToSchemaElement<>(RIM_FACTORY::createRegistryObjectType);

    public Optional<RegistryObjectType> convertRegistryObject(Map<String, Object> map) {
        // ...
        mapToSchemaElement.populateStringElement(map, HOME_KEY, optionalRegistryObject, value -> optionalRegistryObject.get().setLid(value));
        // ...
    }
}
```